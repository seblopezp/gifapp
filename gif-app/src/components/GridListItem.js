import React from "react";

export const GridListItem = ({ id, title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
