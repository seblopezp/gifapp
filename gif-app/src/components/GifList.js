import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GridListItem } from "../components/GridListItem";
export const GifList = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">Loading...</p>
      )}
      <div className="card-grid">
        {images.map((img) => (
          <GridListItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
