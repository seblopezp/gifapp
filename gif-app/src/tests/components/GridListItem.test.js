import React from "react";
import { shallow } from "enzyme";
import { GridListItem } from "../../components/GridListItem";
const title = "Un titulo";
const url = "https://localhost/algo.jpg";
const wrapper = shallow(<GridListItem title={title} url={url} />);

describe("Pruebas en GridItem", () => {
  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe ser una imagen, ser igual a la url y tener tener alt", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe de tener animate_fadeIn", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
