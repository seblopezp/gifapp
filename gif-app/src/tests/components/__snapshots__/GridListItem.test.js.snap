// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p
    className="subtitle is-6"
  >
    Un titulo
  </p>
</div>
`;
